lib/ansible/playbook/role_include.py

```diff
@@ -137,7 +138,10 @@ def load(data, block=None, role=None, task_include=None, variable_manager=None,
         # build options for role includes
         for key in my_arg_names.intersection(IncludeRole.FROM_ARGS):
             from_key = key.replace('_from', '')
-            ir._from_files[from_key] = basename(ir.args.get(key))
+            args_value = ir.args.get(key)
+            if not isinstance(args_value, string_types):
+                raise AnsibleParserError('Expected a string for %s but got %s instead' % (key, type(args_value)))
+            ir._from_files[from_key] = basename(args_value)
```

Comment:
I think this is acceptable for now, but for the future, I think we should see if we can make use of post validating to cover all arguments in a single standard way.